File: fix_delete.js
const fs = require('fs');
let code = fs.readFileSync('app/src/main/java/com/example/ui/viewmodel/JuktiViewModel.kt', 'utf8');

const regex = /val db = com\.example\.JuktiApplication\.getFirestore\(getApplication\(\)\) \?\: return false[\s\S]*?repository\.deleteUserAccount\(uid, userEmail\)/;

const replacement = `val data = mapOf("targetEmail" to userEmail.trim())
            val functions = com.google.firebase.functions.FirebaseFunctions.getInstance()
            functions.getHttpsCallable("deleteUserCompletely").call(data).await()
            
            repository.deleteUserAccount(uid, userEmail)`;

code = code.replace(regex, replacement);
fs.writeFileSync('app/src/main/java/com/example/ui/viewmodel/JuktiViewModel.kt', code);

File: fix_delete.py
import re

with open('app/src/main/java/com/example/ui/viewmodel/JuktiViewModel.kt', 'r') as f:
    content = f.read()

delete_questions = """    fun deleteQuestions(questions: List<QuestionEntity>) {
        logActivity("Deleted ${questions.size} questions")
        viewModelScope.launch {
            questions.forEach { q ->
                repository.deleteQuestion(q)
            }
            _syncToastMessage.value = "Successfully deleted ${questions.size} questions."
        }
    }"""

content = content.replace('    fun deleteQuestion(question: QuestionEntity) {', delete_questions + '\n\n    fun deleteQuestion(question: QuestionEntity) {')

with open('app/src/main/java/com/example/ui/viewmodel/JuktiViewModel.kt', 'w') as f:
    f.write(content)
